src/main/java/com/commercetools/sync/customobjects/CustomObjectSync.java

```diff
@@ -0,0 +1,329 @@
+                    .orElseGet(() -> applyCallbackAndCreate(newCustomObjectDraft));
+            })
+            .map(CompletionStage::toCompletableFuture)
+            .toArray(CompletableFuture[]::new));
+    }
+
+    /**
+     * Given a custom object draft, this method applies the beforeCreateCallback and then issues a create request to the
+     * CTP project to create the corresponding CustomObject.
+     *
+     * @param customObjectDraft the custom object draft to create the custom object from.
+     * @return a {@link CompletionStage} which contains an empty result after execution of the create.
+     */
+    @Nonnull
+    private CompletionStage<Optional<CustomObject<JsonNode>>> applyCallbackAndCreate(
+        @Nonnull final CustomObjectDraft<JsonNode> customObjectDraft) {
+
+        return syncOptions
+            .applyBeforeCreateCallback(customObjectDraft)
+            .map(draft -> customObjectService.upsertCustomObject(customObjectDraft)
+                                             .thenApply(customObjectOptional -> {
+                                                 if (customObjectOptional.isPresent()) {
+                                                     statistics.incrementCreated();
+                                                 } else {
+                                                     statistics.incrementFailed();
+                                                 }
+                                                 return customObjectOptional;
+                                             })
+            )
+            .orElse(CompletableFuture.completedFuture(Optional.empty()));
+    }
+
+
+    /**
+     * Given an existing {@link CustomObject} and a new {@link CustomObjectDraft}, the method verifies whether the JSON
+     * objects between {@link CustomObject#getValue()} and {@link CustomObjectDraft#getValue()} are identical. If so, a
+     * request is made to CTP to update the existing custom object, otherwise it doesn't issue a request.
+     *
+     * <p>The {@code statistics} instance is updated accordingly to whether the CTP request was carried
+     * out successfully or not. If an exception was thrown on executing the request to CTP,the error handling method
+     * is called.
+     *
+     * @param oldCustomObject existing custom object that could be updated.
+     * @param newCustomObject draft containing data that could differ from data in {@code oldCustomObject}.
+     * @return a {@link CompletionStage} which contains an empty result after execution of the update.
+     */
+    @Nonnull
+    private CompletionStage<Optional<CustomObject<JsonNode>>> updateCustomObject(
+        @Nonnull final CustomObject<JsonNode> oldCustomObject,
+        @Nonnull final CustomObjectDraft<JsonNode> newCustomObject) {
+
+        if (CustomObjectSyncUtils.hasIdenticalValue(oldCustomObject, newCustomObject)) {
+            return customObjectService
```

Comment:
this looks wrong to me, we should not update if the resource values are identical, right? 
we should do nothing 🤔 Also in java doc the description looks wrong.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/569/commits/dd28a08ef31828aacff128aea62ee494ba97ad14

Reply:
Fixed related javadoc
https://github.com/commercetools/commercetools-sync-java/pull/569/commits/cd01acebcdb4ba1ca7dfb54e05ccddbc4e4b3b2f